account_avatax_exemption_sign_oca/models/exemption.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 ForgeFlow S.L. (http://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ResPartnerExemption(models.Model):
+    _inherit = "res.partner.exemption"
+
+    sign_oca_request_id = fields.Many2one("sign.oca.request")
+
+    @api.depends("state")
+    def _cancel_sign_request_id(self):
+        for rec in self:
+            if rec.state == "cancel":
+                rec.sign_oca_request_id.cancel()
```

Comment:
```suggestion
	def write(self, vals):
		if vals.get("state") == "cancel":
			self.sign_oca_request_id.cancel()
		res = super().write(vals)
```
@api.depends only would be triggered to compute fields (don't apply in this field), so I suggest changing the logic to handle cases on write method.

Reply:
thanks, I added this module in a separate PR.

here I added it only because we wanted to test sign exemption and the module for exemption is not merged in 16.0 branch.

we tried to add it from the PR in test requirements as dependency but didn't succeed.

we can resume discussion about exemption module in its open PR for 16.0 branch.